addon/mixins/adapter-fetch.js

```diff
@@ -165,7 +169,20 @@ export function determineBodyPromise(response, requestData) {
 }
 
 export default Mixin.create({
-  adapter: service(),
+  fastboot: computed(function() {
+    let owner = getOwner(this);
+    return owner && owner.lookup('service:fastboot');
+  }),
+
+  protocol: computed(function() {
```

Comment:
why use CPs with dependent keys? Can these be set in `init`?

Reply:
@chrism Yea you're absolutely right, thansk for pointing out!